### PR TITLE
Moving to controller runtime

### PR DESCRIPTION
**Pull request by lawrencegripper**
Current status: ~~Broken~~ Working (more testing needed)

### CODE REVIEW

**Comment by lawrencegripper** [comment]
Looking at providing #7

**Comment by lawrencegripper** [comment]
BCD86E6B-BE92-433E-BB70-4A5FFF10EFA8

Looks like I need to set the group version kind on the unstructured resource then the client will hopefully work... fingers crossed.

**Comment by lawrencegripper** [comment]
That did the trick, controller runtime registering a generic controller for each of the resources and handling reconciliation. The controller runtime client also working for update calls to edit the CRD!

@stuartleeks going to have more of a play tomorrow morning while waiting on some other terraform work. You ok to have a review - if you happy I'll remove the current `informer.go` code as this feels much nicer.

**Comment by lawrencegripper** [comment]
image

I've update the logic so errors are pushed back to the controller runtime correctly and added handling of the `notfound` case. Integration tests pass with the controller runtime in place :)

File: controller.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+		return ctrl.Result{}, fmt.Errorf("Failed TF Reconciler on resource %q %w", req.NamespacedName.String(), err)
+	}
+
+	err = r.Client.Update(ctx, resource)
+	if err != nil {
+		log.Error(err, "Failed saving resource")
+		return ctrl.Result{}, fmt.Errorf("Failed saving resource %q %w", req.NamespacedName.String(), err)
+	}
+
+	// Detect drift by checking resource every x mins
```
Comment (stuartleeks) [inline controller.go:72]: yay!

**Pull request merged**